### PR TITLE
Implement CPU time limit for symbolic execution/constraint solver

### PR DESCRIPTION
**Pull request by claudemarche**

### CODE REVIEW

File: lib/constraints/src/implementation/naive/engine.ml [hunk 1]
```diff
@@ -58,6 +58,7 @@ let normalize ?(limit=50) (disj : Disj.t) : Disj.t =
 
 let simplify (disj : Disj.t) : Disj.t =
   Log.debug (fun m -> m "Simplifying");
+  Log.check_cpu_time_limit ();
```
Comment (benozol) [inline lib/constraints/src/implementation/naive/engine.ml]: That seems like a good location to check, indeed! BTW I am working on changing codomain of the variable environment to pairs of strings and export flags – I hope to send a PR before the evening

Comment (Niols) [reply lib/constraints/src/implementation/naive/engine.ml]: Yes, but why in `Log`?

Comment (claudemarche) [reply lib/constraints/src/implementation/naive/engine.ml]: It was not easy for me to understand the modules structure in lib/constraints. Feel free to move it somewhere else.

**Comment by Niols** [comment]
Fair enough!

**Pull request merged**